ethportal-api/src/lib.rs

```diff
@@ -10,16 +10,18 @@ mod web3;
 
 pub use crate::discv5::{Discv5ApiClient, Discv5ApiServer};
 pub use history::{HistoryNetworkApiClient, HistoryNetworkApiServer};
-pub use types::content_item::{
-    BlockBody, ContentItem, EpochAccumulator, HeaderRecord, HeaderWithProof, HistoryContentItem,
-};
-pub use types::content_key::{
+pub use web3::{Web3ApiClient, Web3ApiServer};
+
+// Re-exports trin-types
+pub use trin_types::content_item::{ContentItem, HistoryContentItem};
+pub use trin_types::content_key::{
     BlockBodyKey, BlockHeaderKey, BlockReceiptsKey, EpochAccumulatorKey, HistoryContentKey,
     OverlayContentKey, StateContentKey,
 };
-pub use web3::{Web3ApiClient, Web3ApiServer};
+pub use trin_types::execution::accumulator::{EpochAccumulator, HeaderRecord};
+pub use trin_types::execution::block_body::{BlockBody, EncodableHeaderList};
+pub use trin_types::execution::header::{Header, HeaderWithProof};
+pub use trin_types::execution::receipts::{Receipt, Receipts};
 
 // Re-exports jsonrpsee crate
 pub use jsonrpsee;
-
-pub use reth_primitives::{Header, Receipt, TransactionSigned};
```

Comment:
is it a concern if we don't export`TransactionSigned` anymore?

Reply:
Ahh, ya good catch. I skipped that one since it was a type that's specific to `reth_primitives`. But, I've gone ahead and re-exported our implementation of `Transaction`. 

I've also checked w/ glados, and they aren't reliant upon `TransactionSigned`. However, they will need to change some of their import strings when they update the version dependency since most of their `OverlayContentKey` imports are using a deeply nested statement rather than the shallow statement. 
eg. `use ethportal_api::types::content_key::OverlayContentKey;` -> `use ethportal_api::OverlayContentKey`.

Also, it seems to me like we'll certainly want to release `trin-types` as a standalone crate sometime in the near future. Since it's an enum, our `Transaction` type alone isn't necessarily that useful without the `LegacyTransaction` / `Eip1559Transaction` / `AccessListTransaction` types.

Reply:
> Also, it seems to me like we'll certainly want to release `trin-types` as a standalone crate sometime in the near future. Since it's an enum, our `Transaction` type alone isn't necessarily that useful without the `LegacyTransaction` / `Eip1559Transaction` / `AccessListTransaction` types.

Hm, this seems to go against the path I thought we were on, that `trin-types` is for internal usage, and `ethportal-api` is for other projects to use.

Reply:
iiuc, the idea behind including these types is that a 3rd party consumer of `ethportal-api` might want access to our core types - eg `Header` / `Transaction`. I've updated this pr to only re-export the content key & content item types. The reason being that these are the only types that are currently consumed by `glados`. 

With regards to the core types - whether we re-export those or force libraries to also depend upon `trin-types` -
 let's continue the conversation in #606